presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java

```diff
@@ -552,13 +555,13 @@ private static FieldDefinition generateGroupedField(ClassDefinition definition,
                 Class<?> type = method.getReturnType();
                 checkArgument(supportedClasses.contains(type), type.getName() + " is not supported");
                 String name = method.getName().substring(3);
-                builder.add(new StateField(name, type, getInitialValue(method)));
+                builder.add(new StateField(name, type, getInitialValue(method), method.getName(), fieldTypes.get(name)));
```

Comment:
`Optional.ofNullable(fieldTypes.get(name))`

Change type of `sqlType` argument in `StateField` constructor to `Optional<Type>`.

I'm sorry that I told you to use your discretion when deciding whether to use `null` or `Optional`. While it is true that one should use discretion here, it's also true we are quite strongly inclined to use `Optional`.